### PR TITLE
[Matrix] change to new C++ PVR interface way

### PR DESCRIPTION
**Pull request by AlwinEsch**
Changes related to xbmc/xbmc#16485.

This make the addon to the new C++ addon interface.

Runtime test was OK.

Hope is mostly OK :smile: this is one of the biggest PVR addons :v:, specially with setting value get is now different to before.

### CODE REVIEW

**Comment by janbar** [comment]
@AlwinEsch  GREAT WORK !
I will test it soon.

File: src/settings.cpp [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+    /* If setting is unknown fallback to defaults */
+    kodi::Log(ADDON_LOG_ERROR, "Couldn't get 'channel_icons' setting, falling back to '%u' as default", DEFAULT_CHANNEL_ICONS);
+    m_bChannelIcons = DEFAULT_CHANNEL_ICONS;
+  }
+
+  /* Read setting "recordinm_icons" from settings.xml */
+  if (!kodi::CheckSettingBoolean("recordinm_icons", m_bRecordingIcons))
+  {
+    /* If setting is unknown fallback to defaults */
+    kodi::Log(ADDON_LOG_ERROR, "Couldn't get 'recordinm_icons' setting, falling back to '%u' as default", DEFAULT_RECORDING_ICONS);
```
Comment (janbar) [inline src/settings.cpp]: recordinm_icons  --> recording_icons

File: src/settings.cpp [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+  else if (settingName == "channel_icons")
+  {
+    kodi::Log(ADDON_LOG_INFO, "Changed Setting 'channel_icons' from %u to %u", m_bChannelIcons, settingValue.GetBoolean());
+    if (m_bChannelIcons != settingValue.GetBoolean())
+      return ADDON_STATUS_NEED_RESTART;
+  }
+  else if (settingName == "recordinm_icons")
+  {
+    kodi::Log(ADDON_LOG_INFO, "Changed Setting 'recordinm_icons' from %u to %u", m_bRecordingIcons, settingValue.GetBoolean());
+    if (m_bRecordingIcons != settingValue.GetBoolean())
```
Comment (janbar) [inline src/settings.cpp]: recordinm_icons --> recording_icons

**Review by janbar — CHANGES_REQUESTED** [review]
I am testing it. For now I found a little typo in settings.cpp.

**Comment by janbar** [comment]
IMO it is OK, and ready to merge after fixing the typo.

**Comment by AlwinEsch** [comment]
May still take a little while to come in. I am not at home for the next few weeks and will also look through all the addons beforehand.

**Comment by janbar** [comment]
I have one issue when I open the box to edit a timer: The type isn't right and it always show "Record manual" instead the real type of the timer. Opening the list to edit it, the selected value is right, so I guess the PVR misses to initialize the right value on opening the editing box.

Capture d’écran du 2020-06-07 18-07-10
Capture d’écran du 2020-06-07 18-08-14

@ksooo, It isn't related to this change, but I catch this bug when testing this.

**Comment by janbar** [comment]
@AlwinEsch , thanks for your work. We have a nice API now. Hope it will be accepted as soon as possible.

**Comment by AlwinEsch** [comment]
Have fixed the typo and after all addons are ready can it in.

**Comment by ksooo** [comment]
> @ksooo, It isn't related to this change, but I catch this bug when testing this.

 I think this is fixed by https://github.com/xbmc/xbmc/pull/18027

**Comment by janbar** [comment]
Thanks @ksooo !

What else, the new API is really a great improvement and that will fix all mistakes we could make using directly C structs. Nice.

**Pull request merged**